autogpts/autogpt/autogpt/agents/agent_member.py

```diff
@@ -0,0 +1,263 @@
+import os
+import copy
+import sys
+from uuid import uuid4
+import asyncio
+import logging
+from typing import Optional
+from pydantic import Field
+from autogpt.agents.utils.prompt_scratchpad import PromptScratchpad
+from autogpt.core.resource.model_providers.schema import AssistantChatMessage
+from autogpt.llm.providers.openai import get_openai_command_specs
+from autogpt.models.action_history import Action
+from forge.sdk.db import AgentDB
+from autogpt.core.resource.model_providers.openai import OpenAIProvider
+from forge.sdk.model import (TaskRequestBody, AgentTask, AgentTaskStatus )
+from autogpt.commands import COMMAND_CATEGORIES
+from autogpt.config.config import ConfigBuilder
+from autogpt.agents.base import BaseAgentSettings
+from autogpt.agents.agent import Agent, AgentConfiguration
+from autogpt.models.command_registry import CommandRegistry
+from autogpt.agents.prompt_strategies.divide_and_conquer import Command, DivideAndConquerAgentPromptConfiguration, DivideAndConquerAgentPromptStrategy
+from autogpt.core.prompting.schema import (
+    ChatMessage,
+    ChatPrompt,
+    CompletionModelFunction,
+)
+info = "-v" in sys.argv
+debug = "-vv" in sys.argv
+granular = "--granular" in sys.argv
+
+logging.basicConfig(
+    level=logging.DEBUG if debug else logging.INFO if info else logging.WARNING
+)
+logger = logging.getLogger(__name__)
+
+class AgentMemberSettings(BaseAgentSettings):
+    config: AgentConfiguration = Field(default_factory=AgentConfiguration)
+    prompt_config: DivideAndConquerAgentPromptConfiguration = Field(
+        default_factory=(
+            lambda: DivideAndConquerAgentPromptStrategy.default_configuration.copy(deep=True)
+        )
+    )
+
+class AgentMember(Agent):
+    
+    id: str
+    role: str
+    initial_prompt: str
+    boss: Optional['AgentMember']
+    recruiter: Optional['AgentMember']
+    tasks: list[AgentTask]
+    members: list['AgentMember']
+    create_agent: bool
+    db: AgentDB
+    group: 'AgentGroup'
```

Comment:
How is group conceptually different than members

Reply:
I added a diagram and some description of the difference between them.